clients/src/main/java/org/apache/kafka/common/record/AbstractRecordBatch.java

```diff
@@ -33,4 +35,14 @@ public boolean isCompressed() {
         return compressionType() != CompressionType.NONE;
     }
 
+    @Override
+    public void setSimplified(boolean isSimplified) {
```

Comment:
Instead of this, I think people should just use a separate method to iterate the simplified version.